catalog/mollie/mollie_helper.php

```diff
@@ -398,4 +401,67 @@ public function log ($transaction_id, $status, $order_id)
 			tep_db_query("INSERT INTO " . self::DB_PAYMENTS_TABLE . " (payment_id, status, osc_order_id) VALUES('" . $transaction_id . "', '" . $status . "', '" . $order_id . "')");
 		}
 	}
+
+	/**
+	 * Updates the order status and sends emails if enabled
+	 *
+	 * @param int $order_id 		 Internal order id
+	 * @param string $payment_status Mollie payment status
+	 */
+	public function update_status($order_id, $payment_status)
+	{
+		$new_status_id = static::get_status_id($payment_status);
+
+		$order_updated = false;
+		$check_status_query = tep_db_query("select customers_name, customers_email_address, orders_status, date_purchased from " . TABLE_ORDERS . " where orders_id = '" . (int)$order_id . "'");
+		$check_status = tep_db_fetch_array($check_status_query);
+		$customer_notified = '0';
+
+		if ( $new_status_id && $check_status['orders_status'] != $new_status_id ) 
+		{
+			if ( $this->get_status_notify($payment_status) ) 
+			{
+				require_once dirname(__FILE__) . "/../admin/includes/languages/english/orders.php";
+				require_once dirname(__FILE__) . "/../admin/includes/filenames.php";
+				$internal_status = $this->get_internal_status_name_from_id($new_status_id);
+				$email = STORE_NAME . "\n" . EMAIL_SEPARATOR . "\n" . EMAIL_TEXT_ORDER_NUMBER . ' ' . $order_id . "\n" . EMAIL_TEXT_INVOICE_URL . ' ' . tep_href_link(FILENAME_CATALOG_ACCOUNT_HISTORY_INFO, 'order_id=' . $order_id, 'SSL') . "\n" . EMAIL_TEXT_DATE_ORDERED . ' ' . tep_date_long($check_status['date_purchased']) . "\n\n" . sprintf(EMAIL_TEXT_STATUS_UPDATE, $internal_status);
+				tep_mail($check_status['customers_name'], $check_status['customers_email_address'], EMAIL_TEXT_SUBJECT, $email, STORE_OWNER, STORE_OWNER_EMAIL_ADDRESS);
+				$customer_notified = '1';
+			}
+
+			// Only update order status when order_status is different then the saved order_status
+			tep_db_query("UPDATE orders SET orders_status = '" . $new_status_id . "', last_modified = now() WHERE orders_id = " . intval($order_id));
+			$this->add_order_history($order_id, $payment_status, $customer_notified);
+		}
+	}
+
+	/**
+	 * Retrieve notify setting attached to status
+	 *
+	 * @param $status_name
+	 * @return bool|true
+	 */
+	protected static function get_status_notify ($status_name)
+	{
+		if (isset(static::$STATUS_NOTIFY[$status_name]) && static::$STATUS_NOTIFY[$status_name])
+		{
+			return static::$STATUS_NOTIFY[$status_name] == "True";
+		}
+
+		// fallback, default: send notification.
+		return true;
```

Comment:
Lennard,

 

Ik heb het allemaal terug gewijzigd. En uiteraard getest. Lijtk nu inderdaad wel beter.

 

Wanneer er een bestelling wordt geplaatst via banktransfer dan wordt er nog geen voorraad afgeschreven, pas wanneer het betaald is. Net zoals bij de andere betaalmethodes, maar wanneer het een banktransfer betreft wordt het op ‘on-hold’ gezet i.p.v. ‘pending’ waardoor het na # minuten niet weer verwijderd wordt, de voorraad wordt dan ook niet weer opgeteld omdat er eigenlijk alleen voorraad afgehaald kan worden wanneer het betaald is. Op zich heel logisch. 

 

Ik heb een falende betaling getest door onderstaand geselecteerde toe te voegen. Bij een ‘cancel’ wordt de order dus niet op cancelled gezet! 

In andere gevallen als ‘expired’ wordt de order dus ook netjes op gefaald gezet :) 

 

Volgens mij werkt het netjes zo. Ik heb het nog niet getest in een eerdere Woo versie, daar test jij het nog even mee begreep ik?

 

Ik heb tevens een PULL request aangemaakt.

 



 

From: Lennard van Gunst [mailto:notifications@github.com] 
Sent: woensdag 15 april 2015 14:22
To: mollie/osCommerce
Cc: George van Engers
Subject: Re: [osCommerce] Added confirmation e-mail (#1)

 

In catalog/mollie/mollie_helper.php <https://github.com/mollie/osCommerce/pull/1#discussion_r28414303> :

> +
> +     /**
> +      * Retrieve notify setting attached to status
> +      *
> +      * @param $status_name
> +      * @return bool|true
> +      */
> +     protected static function get_status_notify ($status_name)
> +     {
> +            if (isset(static::$STATUS_NOTIFY[$status_name]) && static::$STATUS_NOTIFY[$status_name])
> +            {
> +                    return static::$STATUS_NOTIFY[$status_name] == "True";
> +            }
> +
> +            // fallback, default: send notification.
> +            return true;

Je zou een .editorconfig kunnen gebruiken. Deze heb ik al eens toegevoegd voor WooComemmerce en PrestaShop. Zie https://github.com/mollie/WooCommerce/blob/master/.editorconfig

Je editor moet wel ondersteuning hebben zodat de .editorconfig wordt gerespecteerd.

Zie http://editorconfig.org/#download voor IDE plugins

—
Reply to this email directly or view it on GitHub <https://github.com/mollie/osCommerce/pull/1/files#r28414303> .  <https://github.com/notifications/beacon/ADrkE7e6lV3qZK52SKlzI7kQrNMc-MSwks5n_k95gaJpZM4D_e1C.gif>